presto-common/src/main/java/com/facebook/presto/common/type/SemanticType.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.QualifiedObjectName;
+
+import static java.util.Objects.requireNonNull;
+
+public class SemanticType
```

Comment:
since it's really just a type + a name, should we call this a `NamedType` instead?

(semantic implies that it's a type that contains info about the meaning of your data, but you could have a  semantic type that is anonymous)

Reply:
another thought: this looks a lot like an early version of what the SQL2016 spec calls a "distinct type"
"The definition of a distinct type specifies its name and the name of its source type." (section 4.7.2)

Reply:
if we ever wanted to support "opaque alias types" like
`Kilogram as bigint` and `gram as bigint`
such that
```
CAST(5 as kilogram) = cast(3 as gram)
```
is a type-checking error, would this design allow for that?

Reply:
Ya I thought about calling it NamedType instead. It kinda depends on whether we want to make this specific to external types, or change this to be something more fundamentally used in parser (`BIGINT` the physical type can also have a corresponding semantic type `BIGINT`).